cabbage/tasks.py

```diff
@@ -9,19 +9,17 @@
 
 class Task:
     def __init__(
-        self, *, manager: "TaskManager", queue: str, name: Optional[str] = None
+        self,
+        func: Callable,
+        *,
+        manager: "TaskManager",
+        queue: str,
+        name: Optional[str] = None,
     ):
         self.queue = queue
         self.manager = manager
-        self.func: Optional[Callable] = None
-        self.name = name
-
-    def __call__(self, func: Callable) -> "Task":
-        self.func = func
-        if not self.name:
-            self.name = func.__name__
-        self.manager.register(self)
-        return self
+        self.func: Optional[Callable] = func
```

Comment:
It's not optional anymore, then, is it ?

Reply:
That's true. I will update it

Reply:
Updated :-)